activerecord/lib/active_record/core.rb

```diff
@@ -534,17 +526,17 @@ def pretty_print(pp)
       return super if custom_inspect_method_defined?
       pp.object_address_group(self) do
         if defined?(@attributes) && @attributes
-          column_names = self.class.column_names.select { |name| has_attribute?(name) || new_record? }
-          pp.seplist(column_names, proc { pp.text "," }) do |column_name|
+          inspect_hash = attributes_for_inspect(self.class.filter_attributes)
```

Comment:
Both `#inspect` and `#pretty_print` need a list of attribute names to display. They don't share handling of the filtered values, however. That suggests extracting `#attribute_names_for_inspect` rather than an `#attributes_for_inspect`, then using a new single-parameter ParameterFilter API here:
```ruby
pp.pp self.class.inspect_filter.filter_param(attr_name, value)
```